tuhi/export.py

```diff
@@ -82,27 +82,44 @@ class JsonSvg(ImageExportBase):
     _base_pen_width = 0.4
     _pen_pressure_width_factor = 0.2
 
+    # Change this value down to reduce size, change it up to improve accuracy. measured in px
+    width_precision = 10
+
     def _convert(self):
 
         width, height = self.output_dimensions
         size = width * mm, height * mm
-        svg = svgwrite.Drawing(filename=self.filename, size=size)
+
+        # Make sure to set viewBox here so mm doesn't have to be specified in all later parts
+        svg = svgwrite.Drawing(filename=self.filename, size=size, viewBox=(f'0 0 {width} {height}'))
 
         g = svgwrite.container.Group(id='layer0')
         for sk_num, stroke_points in enumerate(self.output_strokes):
-            lines = svgwrite.container.Group(id=f'sk_{sk_num}', stroke='black')
+            path = None
+            stroke_width_p = None
             for i, (x, y, stroke_width) in enumerate(stroke_points):
-                if i != 0:
-                    xp, yp, stroke_width_p = stroke_points[i - 1]
-                    lines.add(
-                        svg.line(
-                            start=(xp * mm, yp * mm),
-                            end=(x * mm, y * mm),
-                            stroke_width=stroke_width,
-                            style='fill:none'
-                        )
-                    )
-            g.add(lines)
+                if not x or not y:
+                    continue
+
+                # Reduce precision of the width
+                stroke_width = (stroke_width * self.width_precision) // self.width_precision
```

Comment:
I just stared at this for a few min... isn't this just `stroke_width = int(stroke_width)`?

This is also the issue with my zero stroke width. The stroke width is a delta related to `_base_pen_width`, so for the SVG case that's `0.4 ± 0.2`. This cannot ever be anything but zero when we `int()` the value.

Previously the value was taken as is, so the SVG output was:
```xml
 <line stroke-width="0.3345703125" .../>
```
If I remove this line, the SVG looks more sensible in inkscape. Or rather, if I restore your original code:

```suggestion
                 # clip to one decimal
                 stroke_width = int(stroke_width * self.width_precision) / self.width_precision
```
because I completely misread what you were doing here (hence the comment now :)

Reply:
Ah yes, so it's supposed to be chopping off the end digits so the comparison isn't wasteful. Making a new path because one line is 0.3322334243 and the next is 0.33225 seems a bit silly. (side note: there is a way to generate a power-stroke object, but that is inkscape-svg and not svg so best not to use that here)

I can revert.